src/commands/schema/diff.ts

```diff
@@ -20,76 +29,107 @@ export default class DiffSchemaCommand extends SchemaCommand {
-        method: "POST",
-        headers: { AUTHORIZATION: `Bearer ${secret}` },
-        body: this.body(files),
-        // @ts-expect-error-next-line
-        duplex: "half",
-      });
-      const json = await res.json();
-      if (json.error) {
-        this.error(json.error.message);
-      }
+        if (status === "none") {
+          this.log("There is no staged schema present.");
+        } else {
+          this.log(statusJson.diff ? statusJson.diff : "No schema differences");
+        }
+      } else {
+        const params = new URLSearchParams({
+          ...(hasColor() ? { color: colorParam() } : {}),
+          staged: source === "staged" ? "true" : "false",
+          ...(version !== undefined ? { version } : { force: "true" }),
+          diff: diffKind,
+        });
+        const res = await fetch(new URL(`/schema/1/validate?${params}`, url), {
+          method: "POST",
+          headers: { AUTHORIZATION: `Bearer ${secret}` },
+          body: this.body(files),
+          // @ts-expect-error-next-line
+          duplex: "half",
+        });
+        const json = await res.json();
+        if (json.error) {
+          this.error(json.error.message);
+        }
 
-      if (status !== "none") {
-        if (this.flags?.active) {
+        if (status === "none") {
           this.log(
-            `Differences between the ${bold()}local${reset()} schema and the ${bold()}remote, active${reset()} schema:`
+            `Differences from the ${bold()}remote${reset()} schema to the ${bold()}local${reset()} schema:`
           );
         } else {
-          this.log(
-            `Differences between the ${bold()}local${reset()} schema and the ${bold()}remote, staged${reset()} schema:`
-          );
+          if (source === "active") {
+            this.log(
+              `Differences from the ${bold()}remote, active${reset()} schema to the ${bold()}local${reset()} schema:`
+            );
+          } else {
+            this.log(
+              `Differences from the ${bold()}remote, staged${reset()} schema to the ${bold()}local${reset()} schema:`
+            );
+          }
         }
-      } else {
-        this.log(
-          `Differences between the ${bold()}local${reset()} schema and the ${bold()}remote${reset()} schema:`
-        );
+        this.log(json.diff ? json.diff : "No schema differences");
       }
-      this.log(json.diff ? json.diff : "No schema differences");
     } catch (err) {
       this.error(err);
     }
   }
+
+  parseTarget(): [Target, Target] {
+    const target: string = this.args?.target;
+
+    if (!target) {
+      return ["staged", "local"];
+    }
+
+    if (target === "active") {
+      return ["active", "local"];
+    } else if (target === "staged") {
+      return ["active", "staged"];
+    } else {
+      // NB: `this.error` quits the program, so return `any` to make typescript happy.
+      this.error("Invalid target. Expected: active or staged");
+      return [] as any;
```

Comment:
if you expand the try/catch block in `run()` to include the `parseTarget()` call, you can throw from here instead of exiting with `this.error` and satisfy the type contract in a more consistent way.